src/BatteryStateOfChargeService.cpp

```diff
@@ -43,21 +48,48 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return totalCurrent_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if (newCurrent_ <= 0)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    return estimatedTime_;
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
-{
-    Q_UNUSED(batteryData);
-    // Update your variables here.
+{   
+
+    QTime newTime = batteryData.time;
+    QTime oldTime = newTime;
```

Comment:
Here `oldTime` and `newTime` will be equal, you probably want to set the `oldTime` before you set `newTime`.